lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -536,7 +536,9 @@ def _run_script(self, rerun_data: RerunData) -> None:
                 self._session_state[SCRIPT_RUN_WITHOUT_ERRORS_KEY] = True
         except RerunException as e:
             rerun_exception_data = e.rerun_data
-            premature_stop = True
+            # Interruption due to a rerun is usually from `st.rerun()`, which
```

Comment:
nit: I'm not sure if it's more likely that we hit one of these due to `st.rerun` or due to a user interacting with a widget before the script has run to completion, so I think it'd be good to mention both scenarios in the comment